src/python/pants/engine/console.py

```diff
@@ -55,8 +60,12 @@ def __init__(
 
         has_scheduler = session is not None
 
-        self._stdout = stdout or (NativeStdOut(session) if has_scheduler else sys.stdout)
-        self._stderr = stderr or (NativeStdErr(session) if has_scheduler else sys.stderr)
+        self._stdout = stdout or (
```

Comment:
We should save it for another PR, but: the SchedulerSession should probably not be optional on this class. Maybe it is used in tests without one...?